docker/Dockerfile.dev

```diff
@@ -0,0 +1,9 @@
+# A Dockerfile that can be used in isolation
```

Comment:
What is Dockerfile.dev for?

Reply:
This comment is a mistake from copy/paste. I added a second commit explaining the use cases of the two Dockerfiles. In short, `Dockerfile` always uses malb/lattice-estimator at HEAD, and `Dockerfile.dev` uses a local copy of the repository, including any un-committed changes.